### ISSUE TITLE
shared_image_gallery_destination fails with invalid subscription ID in request

### ISSUE THREAD

**Issue by neildomino** [hashicorp/packer#10746]
#### Overview of the Issue

I'm having trouble using shared_image_gallery_destination in azure-arm builder. I can publish an image without that but when I add it I see an error which says 
"the Shared Gallery Image to which to publish the managed image version to does not exist in the resource group x"

#### Reproduction Steps

packer build packer.json

### Packer version

1.7.0

### Simplified Packer Buildfile

```
            "shared_image_gallery_destination": {
                "resource_group": "NHRG",
                "gallery_name": "nhimagegallery",
                "image_name": "nhtestimage",
                "image_version": "0.0.1",
                "replication_regions": ["West Europe"]
            },
```
### Operating system and Environment details

Linux Ubuntu 18.04 in Azure.

### Log Fragments and crash.log files

```
2021/03/10 14:35:53 packer-builder-azure-arm plugin: Azure request method="GET" request="https://management.azure.com/subscriptions/<subscription_id>/resourceGroups/nhtest2/providers/Microsoft.Compute/images/nhtestimagea?api-version=2018-04-01" body=""
2021/03/10 14:35:53 packer-builder-azure-arm plugin: Azure response status="404 Not Found" method="GET" request="https://management.azure.com/subscriptions/<subscription_id>/resourceGroups/nhtest2/providers/Microsoft.Compute/images/nhtestimagea?api-version=2018-04-01" x-ms-request-id="d2de3950-be38-4d79-b7e1-d81e8caaeae8" body="{\"error\":{\"code\":\"ResourceNotFound\",\"message\":\"The Resource 'Microsoft.Compute/images/nhtestimagea' under resource group 'nhtest2' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix\"}}"
2021/03/10 14:35:53 packer-builder-azure-arm plugin: Azure response status="404 Not Found" method="GET" request="https://management.azure.com/subscriptions/<subscription_id>/resourceGroups/nhtest2/providers/Microsoft.Compute/images/nhtestimagea?api-version=2018-04-01" x-ms-request-id="d2de3950-be38-4d79-b7e1-d81e8caaeae8" body="{\"error\":{\"code\":\"ResourceNotFound\",\"message\":\"The Resource 'Microsoft.Compute/images/nhtestimagea' under resource group 'nhtest2' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix\"}}"
==> azure-arm: WARNING: Zone resiliency may not be supported in West Europe, checkout the docs at https://docs.microsoft.com/en-us/azure/availability-zones/
2021/03/10 14:35:53 packer-builder-azure-arm plugin: Azure request method="GET" request="https://management.azure.com/subscriptions//resourceGroups/NHRG/providers/Microsoft.Compute/galleries/nhimagegallery/images/nhtestimage?api-version=2019-03-01" body=""
2021/03/10 14:35:53 packer-builder-azure-arm plugin: Azure response x-ms-request-id="1dd49d43-3e1d-4836-983a-dedccf066163" body="{\"error\":{\"code\":\"InvalidSubscriptionId\",\"message\":\"The provided subscription identifier 'resourceGroups' is malformed or invalid.\"}}" status="400 Bad Request" method="GET" request="https://management.azure.com/subscriptions//resourceGroups/NHRG/providers/Microsoft.Compute/galleries/nhimagegallery/images/nhtestimage?api-version=2019-03-01"
2021/03/10 14:35:53 packer-builder-azure-arm plugin: Azure response x-ms-request-id="1dd49d43-3e1d-4836-983a-dedccf066163" body="{\"error\":{\"code\":\"InvalidSubscriptionId\",\"message\":\"The provided subscription identifier 'resourceGroups' is malformed or invalid.\"}}" status="400 Bad Request" method="GET" request="https://management.azure.com/subscriptions//resourceGroups/NHRG/providers/Microsoft.Compute/galleries/nhimagegallery/images/nhtestimage?api-version=2019-03-01"
```

Notice ".../subscriptions//resourceGroups/..." in each of the galleries requests. Prior requests to Azure contained my subscription ID in this place.

**Comment by hidalgopl** [comment]
I'm facing the same issue. If i try to add `subscription` into `"shared_image_gallery_destination": {...}` I'm getting:
```
1 error occurred:
	* unknown configuration key: "shared_image_gallery_destination.subscription"; raws is []interface ***map[string]interface ***"client_id":"***", "client_secret":"***", "image_offer":"UbuntuServer", "image_publisher":"Canonical", "image_sku":"16.04-LTS", "location":"westus2", "managed_image_name":"elotl-kipdev-v1.0.7-7-g2d37ce1", "managed_image_resource_group_name":"elotl-resources", "os_type":"Linux", "shared_image_gallery_destination":map[interface ***]interface ***"gallery_name":"kipdev", "image_name":"elotl-kipdev", "image_version":"v1.0.7-7-g2d37ce1", "replication_regions":[]interface ***"eastus", "westus2"***, "resource_group":"elotl-resources", "subscription":"***"***, "subscription_id":"***", "tenant_id":"***", "vm_size":"Standard_B1s_v2"***, map[string]interface ***"packer_build_name":"azure-arm", "packer_builder_type":"azure-arm", "packer_debug":true, "packer_force":false, "packer_on_error":"", "packer_template_path":"/github/workspace/packer.json", "packer_user_variables":map[interface ***]interface ***"aws_access_key":"***", "aws_region":"us-east-1", "aws_secret_key":"***", "azure_client_id":"***", "azure_client_secret":"***", "azure_region":"westus2", "azure_subscription_id":"***", "azure_tenant_id":"***", "gce_account_file":"account.json", "gce_project_id":"elotl-kip", "gce_zone":"us-west1-a", "package":"kip-cell_1.0.7-7-g2d37ce1_amd64.deb", "version":"v1.0.7-7-g2d37ce1"*** 

 and ctx data is map[interface ***]interface ***(nil)
 ```

I dig for a while in the `builders/azure` and I think this line produces false positives:
https://github.com/hashicorp/packer/blob/master/builder/azure/arm/config_test.go#L1952-L1953

I guess it should be changed to `if err != nil { ... }`.
Anyway, seems that `shared_image_gallery_destination` is not working either way.
If someone can point me a place where this `unknown configuration key: "shared_image_gallery_destination.subscription"` error is produced, I'll be happy to fix it.

**Comment by sylviamoss** [comment]
It seems that this issue is similar to https://github.com/hashicorp/packer/issues/10428. Could some of you try our nightly pre-release and let me know if this still happens? 
Here the binaries you can try: https://github.com/hashicorp/packer/releases/tag/nightly

**Comment by neildomino** [comment]
Hi Sylvia,

Thanks, that version is doing what I expected and I can see the subscription ID being included in the URL now. Any time frame for release? Or do I just keep a lookout for 1.7.2?

Neil

**Comment by sylviamoss** [comment]
Good to hear that it worked out for you! Thanks for testing. We should release 1.7.2 in two or three weeks. You should definitely keep a lookout in case we release it in a different timeframe.

**Comment by sylviamoss** [comment]
Sorry, I got the version numbers wrong. This will be released with 1.7.1

**Comment by hidalgopl** [comment]
thanks :+1:
